mcf-browser/src/TimelineExplorer.tsx

```diff
@@ -0,0 +1,390 @@
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Series[]} series that match the filter
+   */
+  getFilteredData(locations: string[]): Series[] {
+    const cleanedLocations = new Set(locations);
+
+    const filteredData = this.props.data.filter(
+        (series) => cleanedLocations.has(series.observationAbout),
+    );
+
+    return filteredData;
+  }
+
+  /** Processes the data passed in by props and returns the
+    * data grouped by variableMeasured
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Grouping} an object mapping from variableMeasured to an array
+    * of Series with that variableMeasured value
+    */
+  getVariableToSeries(locations: string[]): Grouping {
+    const output: Grouping = {};
+    // Filter by location
+    const data = this.getFilteredData(locations);
+
+    for (const series of data) {
+      const varMeasured = series.variableMeasured ?
+        series.variableMeasured :
+        '';
+      if (!output[varMeasured]) {
+        output[varMeasured] = [];
+      }
+      output[varMeasured].push(series);
+    }
+
+    return output;
+  }
+
+
+  /** Returns the JSX to render a group of related series
+    * @param {Series[]} seriesList a list of series objects with
+    *                              the same varMeasured
+    * @param {LocationMapping} locationMapping a mapping from location dcid to
+    * name
+    * @return {JSX.Element} a details element plotting all of the series
+   */
+  renderSeriesGroup(seriesList: Series[], locationMapping: LocationMapping)
+    : JSX.Element {
+    const varMeasured = seriesList[0].variableMeasured;
+    const groups = groupSeriesByLocations(seriesList);
+    const groupNames = Object.keys(groups);
+
+    return (
+      <details className="stat-var-section" key={varMeasured}>
+        <summary>{varMeasured}</summary>
+        {
+          groupNames.map(
+              (groupName) =>
+                renderTimeGraph(
+                    groups[groupName],
+                    groupName,
+                    groupNames.length === 1,
+                    locationMapping,
+                ),
+          )
+        }
+      </details>
+    );
+  }
+
+  /** Get all locations using the observationAbout property
+    * @return {SelectOption[]} a list of unique location objects
+    */
+  private async getAllLocations(): Promise<SelectOption[]> {
+    // Get an array of unique locations
+    const seenLocations = new Set<string>();
+    const locations: string[] = [];
+    for (const series of this.props.data) {
+      const location = series.observationAbout;
+      if (!seenLocations.has(location)) {
+        locations.push(location);
+        seenLocations.add(location);
+      }
+    }
+
+    const labelPromises = locations.map(async (location) => {
+      const label =
+      (location.startsWith('dcid:') ?
+        await getName(location.slice(5)) :
```

Comment:
i don't think the async or await are needed here? because we're just getting the promises & the waiting will be done with the Promise.all in lilne 244

Reply:
Changed!